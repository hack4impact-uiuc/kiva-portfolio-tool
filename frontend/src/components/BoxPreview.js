import React, { Component } from 'react'

import { ContentPreview } from 'box-ui-elements'
import messages from 'box-ui-elements/i18n/en-US'
import 'box-ui-elements/dist/preview.css'
import '../styles/index.scss'

import { getAccessToken } from '../utils/ApiWrapper'

const container = document.querySelector('.container')
const language = 'en-US'
const FILE_ID_VIDEO = '308566420378'
const FILE_ID_3D = '319004423111'
const FILE_ID_TEXT = '308349801521'
const FILE_ID_EXCEL = '319011536090'
const FILE_ID_AUDIO = '308566419514'
const FILE_ID_IMAGE = '308345646235'

class BoxPreview extends Component {
  constructor(props) {
    super(props)
    this.state = {
      fileId: 419075737331,
      accessToken: null
    }
  }

  async componentDidMount() {
    const res = await getAccessToken()
    if (res) {
      this.setState({
        accessToken: res
      })
    } else {
      this.setState({
        accessToken: null
      })
    }
  }

  render() {
    return (
      <ContentPreview
        hasHeader
        fileId={this.state.fileId}
        token={'xNT24dYkz8KM4QBdUnPTLvm3mAxWbgXN'}
        //token={this.state.accessToken}
        language={language}
        messages={messages}
        collection={[
          this.state.fileId,
          419072325627,
            418980052676
        ]}
      />
    )
  }
}

export default BoxPreview
