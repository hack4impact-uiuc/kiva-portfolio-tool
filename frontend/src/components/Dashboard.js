import React, { Component } from 'react'
import { Container, Row, Col, Button } from 'reactstrap'

import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import {
  updateDocuments,
  updateMessages,
  updateInstructions,
  setUserType,
  beginLoading,
  endLoading
} from '../redux/modules/user'

import WithAuth from './auth/WithAuth'
import DocumentList from './DocumentList'
import NavBar from './NavBar'

import {
  getDocumentsByUser,
  getMessagesByFP,
  updateFieldPartnerStatus,
  getFPByID
} from '../utils/ApiWrapper'

import add from '../media/add.png'

const mapStateToProps = state => ({
  isPM: state.user.isPM,
  documents: state.user.documents,
  messages: state.user.messages,
  information: state.user.information
})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      updateDocuments,
      updateMessages,
      setUserType,
      beginLoading,
      endLoading,
      updateInstructions
    },
    dispatch
  )
}
export class Dashboard extends Component {
  constructor(props) {
    super(props)

    this.state = {
      fp_statuses: ['Missing', 'Rejected', 'Pending', 'Approved'],
      pm_statuses: ['Pending', 'Missing', 'Rejected', 'Approved'],
      pm_id: null
    }

    this.handleFinish = this.handleFinish.bind(this)
  }

  async componentDidMount() {
    /**
     * Contains all documents received from backend
     */

    this.props.beginLoading()
    let documentsReceived = []

    // temporary - REMOVE after auth integration
    documentsReceived = await getDocumentsByUser(this.props.match.params.id)
    this.props.setUserType(this.props.match.params.user === 'pm')

    /**
     * Contains all messages received from backend
     */
    const messagesReceived = await getMessagesByFP(
      this.props.match.params.id,
      this.props.match.params.user === 'fp'
    )

    /**
     * Contains all information received from backend
     */
    const fp = await getFPByID(this.props.match.params.id)
    const instructionsReceived = fp.instructions
    this.setState({ pm_id: fp.pm_id })

    if (documentsReceived) {
      this.props.updateDocuments(documentsReceived)
    } else {
      this.props.updateDocuments([])
    }

    if (messagesReceived) {
      this.props.updateMessages(messagesReceived)
    } else {
      this.props.updateMessages([])
    }

    if (instructionsReceived) {
      this.props.updateInstructions(instructionsReceived)
    } else {
      this.props.updateInstructions('')
    }
    this.props.endLoading()
  }

  /**
   * When a Field Partner has finished the process, this method is called to move their status to 'Complete'
   */
  async handleFinish() {
    this.props.beginLoading()
    await updateFieldPartnerStatus(this.props.match.params.id, 'Complete')
    this.props.history.push('/overview/' + this.state.pm_id)
    this.props.endLoading()
  }

  pStyle = {
    margin: 'auto'
  }

  render() {
    return (
      <div className="background-rectangles maxheight">
        <NavBar inDashboard />
        <Container id="dashboard-container">
          {this.props.isPM ? (
            <Row>
              <Col className="text-centered" md="12">
                <Button
                  className="add-doc-text"
                  color="transparent"
                  onClick={() => this.props.history.push('/setup/' + this.props.match.params.id)}
                >
                  <img className="addImg" src={add} alt="Add icon" />
                  <span className="add-doc-text">Update requirements/instructions</span>
                </Button>
              </Col>
              <Col className="text-centered" md="12">
                <Button color="success" onClick={this.handleFinish}>
                  Finish Process
                </Button>
              </Col>
            </Row>
          ) : null}
          <Row>
            {this.props.documents
              ? this.props.isPM
                ? this.state.pm_statuses.map(key => {
                    return (
                      <Col sm="12" md="6" className="dashboard-width-override">
                        <DocumentList documents={this.props.documents[key]} status={key} />
                      </Col>
                    )
                  })
                : this.state.fp_statuses.map(key => {
                    return (
                      <Col sm="12" md="6" className="dashboard-width-override">
                        <DocumentList documents={this.props.documents[key]} status={key} />
                      </Col>
                    )
                  })
              : null}
          </Row>
        </Container>
      </div>
    )
  }
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WithAuth(Dashboard))
