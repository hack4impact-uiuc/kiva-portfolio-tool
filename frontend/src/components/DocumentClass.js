import React, { Component } from 'react'
import { connect } from 'react-redux'
import DocumentClassPreview from './DocumentClassPreview'
import { Button, Modal, ModalBody, ModalFooter } from 'reactstrap'
import {
  deleteDocumentClass,
  updateDocumentClass,
  getAllDocumentClasses
} from '../utils/ApiWrapper'
import { bindActionCreators } from 'redux'
import { updateDocumentClasses } from '../redux/modules/user'
import { beginLoading, endLoading } from '../redux/modules/auth'
import Dropzone from 'react-dropzone'

import '../styles/documentclasspage.css'

import edit from '../media/greyEdit.png'
import remove from '../media/remove.png'

const mapStateToProps = state => ({})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      updateDocumentClasses,
      beginLoading,
      endLoading
    },
    dispatch
  )
}

/*
Corresponds to each individual Document Class in the Admin/PM overview page
Name is represented normally
Buttons exist to view, edit, and delete
View shows the description and a Box preview
*/
export class DocumentClass extends Component {
  constructor(props) {
    super(props)

    this.state = {
      name: this.props.documentClass.name,
      description: this.props.documentClass.description,
      editModal: false, // modal that appears when editing a Document class
      deleteModal: false, // modal that appears making sure the user actually wants to delete the document class
      files: []
    }

    this.editToggle = this.editToggle.bind(this)
    this.deleteToggle = this.deleteToggle.bind(this)
    this.handleDelete = this.handleDelete.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  editToggle() {
    this.setState({ editModal: !this.state.editModal })
  }

  deleteToggle() {
    this.setState({ deleteModal: !this.state.deleteModal })
  }

  updateName = event => {
    this.setState({ name: event.target.value })
  }

  updateDescription = event => {
    this.setState({ description: event.target.value })
  }

  async handleSubmit() {
    this.props.beginLoading()
    if (this.state.files.length == 0) {
      await updateDocumentClass(
        this.props.documentClass._id,
        this.state.name,
        this.state.description,
        null,
        null
      )
    } else {
      await updateDocumentClass(
        this.props.documentClass._id,
        this.state.name,
        this.state.description,
        this.state.files[0],
        this.state.files[0].name
      )
    }
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
    this.props.endLoading()
    this.editToggle()
  }

  onDrop(files) {
    this.setState({
      files
    })
  }

  async handleDelete() {
    this.props.beginLoading()
    await deleteDocumentClass(this.props.documentClass._id)
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
    this.props.endLoading()
    this.deleteToggle()
  }

  render() {
    return (
      <>
        <Modal isOpen={this.state.editModal} toggle={this.editToggle}>
          <ModalBody>
            <form>
              <span> Name: </span>
              <input onChange={this.updateName} value={this.state.name} />
              <br />
              <span> Description: </span>
              <textarea
                name="paragraph_text"
                cols="50"
                rows="10"
                onChange={this.updateDescription}
                value={this.state.description}
              />
              <br />
              <div className="dropPage">
                <section className="droppedBox">
                  <div className="dropZone">
                    <Dropzone onDrop={this.onDrop.bind(this)}>
                      {({ getRootProps, getInputProps }) => (
                        <section>
                          <div {...getRootProps()}>
                            <input {...getInputProps()} />
                            {this.state.files.length > 0 ? (
                              <p>File uploaded: {this.state.files[0].name}</p>
                            ) : (
                              <p>Click to Upload</p>
                            )}
                          </div>
                        </section>
                      )}
                    </Dropzone>
                  </div>
                </section>
                <hr />
              </div>
            </form>
          </ModalBody>
          <ModalFooter>
            <Button className="invalidSearchButton" onClick={this.editToggle}>
              Return
            </Button>
            <Button onClick={this.handleSubmit}>Update Document Class</Button>
          </ModalFooter>
        </Modal>
        <tr className="hoverable">
          {this.props.documentClass.name ? (
            <td data-testid="docClass">{this.props.documentClass.name}</td>
          ) : null}
          <td data-testid="interaction" className="interaction">
            <DocumentClassPreview documentClass={this.props.documentClass} />
            <Button color="transparent" onClick={this.editToggle}>
              <img className="buttonimg" src={edit} />
            </Button>
            <Button color="transparent" onClick={this.deleteToggle}>
              <img className="buttonimg" src={remove} />
            </Button>
            <Modal isOpen={this.state.deleteModal} toggle={this.deleteToggle}>
              <ModalBody>
                <p>
                  Are you sure you want to delete {this.props.documentClass.name}? This will delete
                  every document of this type for all users.
                </p>
              </ModalBody>
              <ModalFooter>
                <Button className="invalidSearchButton" onClick={this.handleDelete}>
                  Delete and return
                </Button>
                <Button className="invalidSearchButton" onClick={this.deleteToggle}>
                  Return
                </Button>
              </ModalFooter>
            </Modal>
          </td>
        </tr>
      </>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DocumentClass)
