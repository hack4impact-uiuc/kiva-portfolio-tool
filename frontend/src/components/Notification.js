import React, { Component } from 'react'
// import DateTimePicker from 'react-datetime-picker';
import { ListGroupItem } from 'reactstrap'
import '../styles/notification.css'
import { connect } from 'react-redux'

const mapStateToProps = state => ({
  isPM: state.user.isPM
})

class Notification extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    const { isPM } = this.props
    return (
      <ListGroupItem>
        <p className="person-name">
          {' '}
          <b> {this.props.name} </b>{' '}
        </p>
        <p className="time"> {this.props.time} </p>
        <p className="notif-description"> {this.props.description} </p>
      </ListGroupItem>
    )
  }
}

export default connect(mapStateToProps)(Notification)
