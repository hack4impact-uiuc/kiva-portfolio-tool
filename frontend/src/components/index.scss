body, html {
    height: 100%;
    width: 100%;
    padding: 0;
    border: 0 none;
    margin: 0;
}

.container {
    height: 100%;
    width: 100%;
}