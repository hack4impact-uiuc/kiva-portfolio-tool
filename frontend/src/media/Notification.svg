<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 46 48.5"><defs><style>.cls-1{fill:#4faf4e;}.cls-2{font-size:39.77px;fill:#fff;font-family:Nunito-Black, Nunito Black;}</style></defs><title>Notification</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><circle class="cls-1" cx="23" cy="23" r="23"/><text class="cls-2" transform="translate(17.59 37.4)">!</text></g></g></svg>