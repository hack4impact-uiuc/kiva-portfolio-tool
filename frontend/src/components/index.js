import Dashboard from './Dashboard'
import DocumentPreview from './DocumentPreview'

import SelectDocumentsPage from './SelectDocuments'
import BoxPreview from './DocumentPreview'
import PMMainPage from './PMMainPage'
import DocumentClassPage from './DocumentClassPage.js'

import Load from './Load'

import ErrorMessage from './auth/ErrorMessage'
import ForgotPassword from './auth/ForgotPassword'
import LogIn from './auth/LogIn'
import Register from './auth/Register'
import WrongPage from './auth/WrongPage'

export {
  Dashboard,
  LogIn,
  BoxPreview,
  Register,
  SelectDocumentsPage,
  PMMainPage,
  ForgotPassword,
  DocumentClassPage,
  DocumentPreview,
  Load,
  WrongPage,
  ErrorMessage
}
