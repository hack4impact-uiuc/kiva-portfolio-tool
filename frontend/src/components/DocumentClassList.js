import React, { Component } from 'react'
import { Table } from 'reactstrap'
import DocumentClass from './DocumentClass'
import '../styles/documentlist.css'

class DocumentClassList extends Component {
  render() {
    return (
      <Table>
        <tbody>
          <tr>
            <th className="theader-centered">Document Class Name</th>
            <th />
          </tr>
          {this.props.documentClasses
            ? this.props.documentClassess.map(documentClass => (
                <DocumentClass documentClass={documentClass} />
              ))
            : null}
        </tbody>
      </Table>
    )
  }
}

export default DocumentClassList
