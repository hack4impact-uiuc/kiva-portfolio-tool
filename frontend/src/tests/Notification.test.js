import React from 'react'
import { Notification } from '../components/Notification'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'

beforeAll(() => {
  configure({ adapter: new Adapter() })
})

test('Notification renders correctly', () => {
  const component = shallow(<Notification />)

  let tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
