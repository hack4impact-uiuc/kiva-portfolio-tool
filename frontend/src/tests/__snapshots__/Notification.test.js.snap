// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
